error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/JooqRules.java

```diff
@@ -0,0 +1,29 @@
+package tech.picnic.errorprone.refasterrules;
+
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+import java.util.stream.Stream;
+import org.jooq.Record;
+import org.jooq.ResultQuery;
+import tech.picnic.errorprone.refaster.annotation.Description;
+import tech.picnic.errorprone.refaster.annotation.OnlineDocumentation;
+
+@OnlineDocumentation
+final class JooqRules {
+  private JooqRules() {}
+
+  /** Prefer eagerly fetching data over (lazy) streaming to avoid potentially leaking resources. */
+  @Description(
+      "Prefer eagerly fetching data over (lazy) streaming to avoid potentially leaking resources.")
```

Comment:
Would it be useful to link to the Jooq article here?

Reply:
While it would be useful, it might not be wise as the blogpost might move URLs one day, and then the code is not correct. I think that would not be prudent in a library such as this. Perhaps linking to it in this PR is enough for curious folks to find one day if they go looking for the motivations for this rule.